### PR TITLE
add environment for deployments on jobs

### PR DESCRIPTION
**Pull request by zeeshanlakhani**
Add support for https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment#using-an-environment. 

@regadas Not sure if I had to do anything else on this?

### CODE REVIEW

**Comment by zeeshanlakhani** [comment]
Also added https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs.

**Comment by zeeshanlakhani** [comment]
@regadas lmk if you have thoughts, etc.

File: defaults/Job.dhall [hunk 1]
```diff
@@ -13,6 +13,7 @@ let Strategy = ../types/Strategy.dhall
 in  { name = None Text
     , needs = None (List Text)
     , strategy = None Strategy
+    , environment = None Text
```
Comment (regadas) [inline defaults/Job.dhall:16]: we could only support the following extended case. wdyt? we could have a JobEnv Type.

```
    environment: 
      name: production
      url: https://github.com
```

File: defaults/Job.dhall [hunk 2]
```diff
@@ -21,4 +22,5 @@ in  { name = None Text
     , services = None (List { mapKey : Text, mapValue : Service })
     , container = None Container
     , concurrency = None Concurrency
+    , permissions = None (List { mapKey : Text, mapValue : Text })
```
Comment (regadas) [inline defaults/Job.dhall:25]: Here we could have a permission type with all the possibilities and an enum type to represent the values `read|write|none`

**Review by regadas — COMMENTED** [review]
Thanks for doing this @zeeshanlakhani

**Comment by zeeshanlakhani** [comment]
@regadas i was being a bit lazy :), but yep, can make these changes easily.

**Comment by zeeshanlakhani** [comment]
@regadas hopefully, this is closer :).

**Comment by zeeshanlakhani** [comment]
@regadas checkin-on this.

**Review by regadas — APPROVED** [review]
@zeeshanlakhani Thanks for this! sorry for the late response I took a few days of vacation 🍹

**Comment by zeeshanlakhani** [comment]
Thanks @regadas!

**Pull request merged**